pkgs/development/tools/misc/binutils/default.nix

```diff
@@ -33,59 +37,80 @@ let
+  normal-src = stdenv.__bootPackages.binutils-unwrapped.src or non-boot-src;
+
+  # Platforms where we directly use the final source.
+  # Generally for cross-compiled platforms, where the boot source won't compile.
+  skipBootSrc = stdenv.targetPlatform.isOr1k;
+
+  # Select the specific source according to the platform in use.
+  src = if stdenv.targetPlatform.isVc4 then vc4-binutils-src
+    else if skipBootSrc then non-boot-src
+    else normal-src;
+
+  patchesDir = ./patches + "/${minorVersion}";
 in
 
 stdenv.mkDerivation {
   pname = targetPrefix + basename;
-  inherit version;
-
-  src = if stdenv.targetPlatform.isVc4 then vc4-binutils-src else normal-src;
+  inherit src version;
 
   patches = [
     # Make binutils output deterministic by default.
-    ./deterministic.patch
+    "${patchesDir}/deterministic.patch"
 
     # Bfd looks in BINDIR/../lib for some plugins that don't
     # exist. This is pointless (since users can't install plugins
     # there) and causes a cycle between the lib and bin outputs, so
     # get rid of it.
-    ./no-plugins.patch
+    "${patchesDir}/no-plugins.patch"
 
     # Help bfd choose between elf32-littlearm, elf32-littlearm-symbian, and
     # elf32-littlearm-vxworks in favor of the first.
     # https://github.com/NixOS/nixpkgs/pull/30484#issuecomment-345472766
-    ./disambiguate-arm-targets.patch
+    "${patchesDir}/disambiguate-arm-targets.patch"
 
     # For some reason bfd ld doesn't search DT_RPATH when cross-compiling. It's
     # not clear why this behavior was decided upon but it has the unfortunate
     # consequence that the linker will fail to find transitive dependencies of
     # shared objects when cross-compiling. Consequently, we are forced to
     # override this behavior, forcing ld to search DT_RPATH even when
     # cross-compiling.
-    ./always-search-rpath.patch
-
-  ] ++ lib.optionals (!stdenv.targetPlatform.isVc4)
-  [
+    "${patchesDir}/always-search-rpath.patch"
+  ]
+  # For version 2.31 exclusively
+  ++ lib.optionals (!stdenv.targetPlatform.isVc4 && minorVersion == "2.31") [
     # https://sourceware.org/bugzilla/show_bug.cgi?id=22868
-    ./gold-symbol-visibility.patch
+    ./patches/2.31/gold-symbol-visibility.patch
 
     # https://sourceware.org/bugzilla/show_bug.cgi?id=23428
     # un-break features so linking against musl doesn't produce crash-only binaries
-    ./0001-x86-Add-a-GNU_PROPERTY_X86_ISA_1_USED-note-if-needed.patch
-    ./0001-x86-Properly-merge-GNU_PROPERTY_X86_ISA_1_USED.patch
-    ./0001-x86-Properly-add-X86_ISA_1_NEEDED-property.patch
-  ] ++ lib.optional stdenv.targetPlatform.isiOS ./support-ios.patch;
+    ./patches/2.31/0001-x86-Add-a-GNU_PROPERTY_X86_ISA_1_USED-note-if-needed.patch
+    ./patches/2.31/0001-x86-Properly-merge-GNU_PROPERTY_X86_ISA_1_USED.patch
+    ./patches/2.31/0001-x86-Properly-add-X86_ISA_1_NEEDED-property.patch
+  ]
+  ++ lib.optional stdenv.targetPlatform.isiOS ./support-ios.patch
```

Comment:
was this one not moved?

Reply:
It wasn't edited in #89793.

I left it as it was as I do not have the knowledge to test it.

We could move it to 2.31.1, and copy into 2.34 without changes, but I'm almost assuming the patch might not apply?